homeassistant/components/delijn/sensor.py

```diff
@@ -37,67 +42,66 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Create the sensor."""
     api_key = config[CONF_API_KEY]
-    name = DEFAULT_NAME
 
     session = async_get_clientsession(hass)
 
     sensors = []
     for nextpassage in config[CONF_NEXT_DEPARTURE]:
-        stop_id = nextpassage[CONF_STOP_ID]
-        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
-        line = Passages(
-            hass.loop, stop_id, number_of_departures, api_key, session, True
+        sensors.append(
+            DeLijnPublicTransportSensor(
+                Passages(
+                    hass.loop,
+                    nextpassage[CONF_STOP_ID],
+                    nextpassage[CONF_NUMBER_OF_DEPARTURES],
+                    api_key,
+                    session,
+                    True,
+                )
+            )
         )
-        await line.get_passages()
-        if line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
-        sensors.append(DeLijnPublicTransportSensor(line, name))
 
     async_add_entities(sensors, True)
 
 
 class DeLijnPublicTransportSensor(Entity):
     """Representation of a Ruter sensor."""
 
-    def __init__(self, line, name):
+    def __init__(self, line):
         """Initialize the sensor."""
         self.line = line
         self._attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._name = name
+        self._name = None
         self._state = None
-        self._available = False
 
     async def async_update(self):
         """Get the latest data from the De Lijn API."""
-        await self.line.get_passages()
-        if self.line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
         try:
+            _LOGGER.debug("updating delijn sensor")
+            await self.line.get_passages()
+
+            self._name = await self.line.get_stopname()
+            _LOGGER.debug("stop name: %s", self._name)
             first = self.line.passages[0]
             if first["due_at_realtime"] is not None:
                 first_passage = first["due_at_realtime"]
             else:
                 first_passage = first["due_at_schedule"]
             self._state = first_passage
-            self._name = first["stopname"]
-            self._attributes["stopname"] = first["stopname"]
+            _LOGGER.debug("state: %s", self._state)
+            self._attributes["is_realtime"] = first["is_realtime"]
```

Comment:
This is a new attribute, right?

Reply:
Yes, it's part of the new pydelijn release, I forgot to add it in the explanation. Should I have made another pull request for this?

Reply:
Yeah, we won't add new features to the release that goes out tomorrow. We can include the fixes. So if you remove this line and make that a separate PR we can tag this PR for the 0.110.0 release.

Reply:
Ok, I'll remove it and create another pull request for that later